### PR TITLE
Add tests for gridui

### PR DESCRIPTION
**Pull request by raju249**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Adds selenium tests for GridUI.
To track - https://github.com/SeleniumHQ/selenium/issues/8756

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [ ] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by AutomatedTester** [comment]
This ia good start. 

* [ ] We need a test that doesn't have a node added, I can see some bugs there already
* [ ] Clicking on a node and checking that we have the page looking correctly
* [ ] if doable, add a test that kills the node and then checks the status has changed (I don't know how long that logic takes to kick in so if it's long ignore it and we need a unit test

**Comment by raju249** [comment]
Thanks! @AutomatedTester 

Point 2 is done.

Point 1 and Point 3 - I will need to start a distributed server rather than a standalone. In the tests that are added here, it works with standalone.

I have added a test for node page and find an element which is only seen on that page and using that to verify the page loads and output.

**Comment by raju249** [comment]
@AutomatedTester - Both tests are up and fixed the `NaN` issue too.

About kicking the node out and checking the updated page, I tried manually, started hub and node and killed the node immediately after, for first 5 mins it did not update in the model, neither I saw logs of the node being marked down, so if its taking more than 5 mins to update the state internally which when read from GQL will update the page, I dont think we should for the test.

What do you think?

**Comment by AutomatedTester** [comment]
> @AutomatedTester - Both tests are up and fixed the `NaN` issue too.
> 
> About kicking the node out and checking the updated page, I tried manually, started hub and node and killed the node immediately after, for first 5 mins it did not update in the model, neither I saw logs of the node being marked down, so if its taking more than 5 mins to update the state internally which when read from GQL will update the page, I dont think we should for the test.
> 
> What do you think?

We will need to create a unit test for that one then.

File: java/server/test/org/openqa/selenium/grid/gridui/ConsolePageTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  @Test
+  public void testNodePage() {
+    Capabilities caps = new ImmutableCapabilities("browserName", "chrome");
+    WebDriver driver = new RemoteWebDriver(server.getUrl(), caps);
+    WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(2));
+
+    driver.get("localhost:" + port + "/ui/index.html#/console");
+    WebElement element = wait.until(ExpectedConditions.visibilityOf(driver.findElement(By.xpath("//a[contains(@href,'node')]"))));
+
+    assertNotNull(element);
```
Comment (AutomatedTester) [inline java/server/test/org/openqa/selenium/grid/gridui/ConsolePageTest.java]: We don't need this assert, `wait` will throw or be successful

**Comment by raju249** [comment]
Yes, I am working on that with unit tests on other components too. I will open up a separate PR which would have those react unit tests.

File: java/server/test/org/openqa/selenium/grid/gridui/ConsolePageTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    element.click();
+
+    WebElement nodePage = wait.until(ExpectedConditions.visibilityOf(driver.findElement(By.xpath("//*[text()='root']"))));
+
+    assertNotNull(nodePage);
```
Comment (AutomatedTester) [inline java/server/test/org/openqa/selenium/grid/gridui/ConsolePageTest.java]: Same here. If we are going to assert lets make it for something that is specific to that page.

File: java/server/test/org/openqa/selenium/grid/gridui/HubConsolePageTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  public void testNoNodePage() {
+    Capabilities caps = new ImmutableCapabilities("browserName", "chrome");
+    WebDriver driver = new RemoteWebDriver(standaloneServer.getUrl(), caps);
+    WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(2));
+
+    driver.get("localhost:" + hubPort + "/ui/index.html#/");
+
+    WebElement element = wait.until(ExpectedConditions.visibilityOf(driver.findElement(By.id("ring-system"))));
+
+    assertNotNull(element);
```
Comment (AutomatedTester) [inline java/server/test/org/openqa/selenium/grid/gridui/HubConsolePageTest.java]: We don't need this assert.

File: java/server/test/org/openqa/selenium/grid/gridui/ConsolePageTest.java (hunk 2 above)
Comment (raju249) [reply java/server/test/org/openqa/selenium/grid/gridui/ConsolePageTest.java]: I removed the assert, and added a unique id to the node info dropdown.

**Review by AutomatedTester — APPROVED** [review]

**Pull request merged**